fortio_main.go

```diff
@@ -536,9 +538,25 @@ func grpcClient() {
 		}
 		return
 	}
-	_, err := fgrpc.PingClientCall(host, count, httpOpts.PayloadUTF8(), *pingDelayFlag, &httpOpts.TLSOptions)
+	_, err := fgrpc.PingClientCall(host, count, httpOpts.PayloadUTF8(), *pingDelayFlag, &httpOpts.TLSOptions, md)
 	if err != nil {
 		// already logged
 		os.Exit(1)
 	}
 }
+
+// httpHeader2grpcMetadata covert md's key to lowercase and filter invalid key.
+func httpHeader2grpcMetadata(headers map[string][]string) map[string][]string {
+	ret := make(map[string][]string)
+	for k, v := range headers {
+		k = strings.ToLower(k)
+		switch k {
+		case "content-length", "content-type":
+			log.Infof("Skipping setting metadata %s:%v", k, v)
```

Comment:
does this trigger using -payload for grpc? (it would if the http options are normalized/processed, because that's where it sets these, if you pass a payload, before this is called)

Reply:
it does:
```
$ go run . load -c 1 -n 3 -payload foo -grpc -ping -H fOo:bAr -H user-agent:foo 127.0.0.1:8079
14:54:58 I http_client.go:288> User-Agent being Set to "foo"
Fortio dev running at 8 queries per second, 10->10 procs, for 3 calls: 127.0.0.1:8079
14:54:58 I fortio_main.go:555> Skipping setting metadata content-type:[application/octet-stream]
14:54:58 I fortio_main.go:555> Skipping setting metadata content-length:[3]
```